src/v/kafka/server/handlers/fetch.cc

```diff
@@ -443,7 +564,9 @@ handle_shard_fetch(ss::shard_id shard, op_context& octx, shard_fetch fetch) {
          deadline = octx.deadline,
          configs = std::move(fetch.requests),
          &replica_selector = octx.rctx.replica_selector(),
-         bytes_left = octx.bytes_left](
+         bytes_left = octx.bytes_left,
+         &memory_sem = octx.rctx.memory_sem_local(),
+         &memory_fetch_sem = octx.rctx.memory_fetch_sem_local()](
```

Comment:
this looks to me like a reference to the semaphores on the calling shard is going to be passed to the destination shard, operating under the assumption that the lambda captures is created on the originating core (which is how i'm imagining that it works). can you confirm that's the case or not?